src/libusermode/uh-private.hpp

```diff
@@ -165,20 +165,21 @@ struct copy_on_write_result_t : public call_result_t
 class userhook : public pluginex
 {
 public:
-    int initialized;
     std::array<size_t, __OFFSET_MAX> offsets;
 
     std::vector<usermode_cb_registration> plugins;
     // map dtb -> list of hooked dlls
     std::map<addr_t, std::vector<dll_t>> loaded_dlls;
 
-    userhook(drakvuf_t drakvuf) : pluginex(drakvuf, OUTPUT_DEFAULT), initialized(0)
-    {
-        drakvuf_get_kernel_struct_members_array_rva(drakvuf, offset_names, __OFFSET_MAX, offsets.data());
-    }
+    userhook(drakvuf_t drakvuf);
```

Comment:
The big brain strat is to declare ctor private, so you are forced to use `get_instance` method.

Reply:
Thanks, nice advice.